components/infra-proxy-service/server/server_test.go

```diff
@@ -0,0 +1,127 @@
+package server
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	healthpb "google.golang.org/grpc/health/grpc_health_v1"
+
+	secrets "github.com/chef/automate/api/external/secrets"
+	authz "github.com/chef/automate/api/interservice/authz/common"
+	authz_v2 "github.com/chef/automate/api/interservice/authz/v2"
+	infra_proxy "github.com/chef/automate/api/interservice/infra_proxy/service"
+	"github.com/chef/automate/lib/grpc/auth_context"
+	"github.com/chef/automate/lib/grpc/grpctest"
+	"github.com/chef/automate/lib/grpc/secureconn"
+	"github.com/chef/automate/lib/logger"
+	"github.com/chef/automate/lib/tls/test/helpers"
+	"github.com/chef/automate/lib/tracing"
+
+	v1 "github.com/chef/automate/components/infra-proxy-service/server/v1"
+	"github.com/chef/automate/components/infra-proxy-service/service"
+	"github.com/chef/automate/components/infra-proxy-service/storage"
+	"github.com/chef/automate/components/infra-proxy-service/storage/postgres/migration"
+)
+
+func TestHealthGRPC(t *testing.T) {
```

Comment:
While running this test case it got failed:

```
Running tool: /usr/local/go/bin/go test -timeout 30s github.com/chef/automate/components/infra-proxy-service/server -run ^(TestHealthGRPC)$

--- FAIL: TestHealthGRPC (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa091a0]

goroutine 7 [running]:
testing.tRunner.func1(0xc0003d2500)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0xaca4a0, 0x11a8950)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/chef/automate/components/infra-proxy-service/server.setupInfraProxyService(0xcbd560, 0xc00003c0f0, 0xc0003d2500, 0xcc8be0, 0xc0000108a8, 0x0, 0x2, 0x0, 0x115aa5c, 0x24, ...)
	/chef-automate/dev/automate/components/infra-proxy-service/server/server_test.go:84 +0x4b0
github.com/chef/automate/components/infra-proxy-service/server.TestHealthGRPC(0xc0003d2500)
	/chef-automate/dev/automate/components/infra-proxy-service/server/server_test.go:34 +0x16b
testing.tRunner(0xc0003d2500, 0xbf1ee8)
	/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x350
FAIL	github.com/chef/automate/components/infra-proxy-service/server	0.013s
FAIL
Error: Tests failed.



```

Reply:
here, definitely I am missing something need someone can look and suggest me some direction?